cpp/ruapu.hpp

```diff
@@ -0,0 +1,39 @@
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#define RUAPU_IMPLEMENTATION
+#include "../ruapu.h"
+
+class ruapu
+{
+public:
+    ruapu() { 
+      static int initialised = 0;
+      if (!initialised) {
+        ruapu_init();
+        initialised = 1;
+      }
+    }
+
+    ~ruapu() = default;
+
+    bool support(std::string isa)
+    {
+        return ruapu_supports(isa.c_str()) == 1;
+    }
```

Comment:
Should be a class method and the argument should be an l-value reference.

```suggestion
    static bool support(const std::string& isa)
    {
        ruapu::init();
        return ruapu_supports(isa.c_str()) == 1;
    }
```